src/app/public/src/modules/table-of-contents/table-of-contents.component.ts

```diff
@@ -9,4 +12,50 @@ import { StacheNavLink } from '../nav';
 export class StacheTableOfContentsComponent {
   @Input()
   public routes: StacheNavLink[];
+  private viewTop: number;
+  private documentBottom: number;
+
+  constructor(
+    private windowRef: StacheWindowRef,
+    private omnibarService: StacheOmnibarAdapterService,
+    private anchorService: StachePageAnchorService
+  ) {
+    this.windowRef.scrollEvent.subscribe(() => {
+      this.updateRoutesOnScroll(this.anchorService.pageAnchors);
+    });
+  }
+
+  public updateRoutesOnScroll(routes: any[]) {
+    if (routes && routes.length) {
+      this.updateView(routes);
+    }
+  }
+
+  // Updates ng-class level trigger to show/hide blue border marking location on page
+  public updateView(routes: BehaviorSubject<StacheNavLink>[]) {
+    this.trackViewTop();
+    this.isCurrent(routes);
+  }
+
+  private trackViewTop() {
+    this.viewTop = (this.windowRef.nativeWindow.pageYOffset + this.omnibarService.getHeight());
+    this.documentBottom = Math.round(this.windowRef.nativeWindow.document.documentElement.getBoundingClientRect().bottom);
+  }
+
+  private isCurrent(routes: BehaviorSubject<StacheNavLink>[]): void {
```

Comment:
Since we want the TOC to remain separate from the page anchors, we should do this filter only on the list it's provided instead of the one from the PageAnchorService.  We could also clean this up a bit to make it easier to read.  Moving the `if(this.windowRef.nativeWindow + 5) >= this.documentBottom) {}` statement out of the forEach loop.  Something like 
```
  private isCurrent(routes: StacheNavLink[]): void {

    if ((this.windowRef.nativeWindow.innerHeight + 5) >= this.documentBottom) {
      routes.forEach((route: StacheNavLink, idx: number) => {
        route.isCurrent = (idx === (routes.length - 1));
      });
      return;
    } 
  
    routes.forEach((route, index) => {
      const nextRoute = routes[index +1];
      if (nextRoute && nextRoute.offsetTop <= this.viewTop) {
        route.isCurrent = false;
        return;
      }

      route.isCurrent = route.offsetTop <= this.viewTop;
    });
  }
```